server/controllers/contentController

```diff
@@ -21,6 +21,49 @@ const createContent = async (req, res) => {
     }
 }
 
+// Search for content
+const searchContent = async (req, res) => {
+    const {query} = req.body     // get the query string
+    try {
+        
+        const results = await content.find({
+            $or: [ //Allows to search simultaneously in several different fields
+                {name:{$regex: query, $options: 'i'}},
+                {workout_type:{$regex: query, $options: 'i'}},
+                {difficulty_level:{$regex: query, $options: 'i'}},
+                {description:{$regex: query, $options: 'i'}}
+            ]
+    }, 'name thumbnail_url description url') //The fields that we will return
+    
+    res.status(200).json(results);
+
+    } catch (err) {
+        res.status(500).json({mssg: 'Error searching content', err});
+    }
+
+};
+
+// Search suggestions for autocomplete
+const searchSuggestions = async (req, res) => {
+    const {query} = req.body; 
+    
+    try{
+        const suggestions = await content.find(
```

Comment:
also - you are allowing to search by multiple fields but presenting suggestions only for one field, this behaviour might be confusing for the user.